internal/pkg/cli/local_run.go

```diff
@@ -106,11 +207,194 @@ func (o *localRunOpts) Execute() error {
 	}
 
 	secrets := taskDef.Secrets()
-	_, err = o.ecsLocalClient.DecryptedSecrets(secrets)
+	decryptedSecrets, err := o.ecsLocalClient.DecryptedSecrets(secrets)
 	if err != nil {
 		return fmt.Errorf("get secret values: %w", err)
 	}
 
+	secretsList := make(map[string]string)
+	for _, s := range decryptedSecrets {
+		secretsList[s.Name] = s.Value
+	}
+
+	envVars := make(map[string]string)
+	envVariables := taskDef.EnvironmentVariables()
+	for _, envVariable := range envVariables {
+		envVars[envVariable.Name] = envVariable.Value
+	}
+
+	containerPorts := make(map[string]string)
+	containerdef := taskDef.ContainerDefinitions
+	for _, container := range containerdef {
+		for _, portMapping := range container.PortMappings {
+			hostPort := aws.Int64Value(portMapping.HostPort)
+			hostPortStr := strconv.FormatInt(hostPort, 10)
+			var containerPort int64
+			if portMapping.ContainerPort == nil {
+				containerPort = hostPort
+			} else {
+				containerPort = aws.Int64Value(portMapping.ContainerPort)
+			}
+			containerportStr := strconv.FormatInt(containerPort, 10)
+			containerPorts[hostPortStr] = containerportStr
+		}
+	}
+
+	envSess, err := o.sessProvider.FromRole(o.targetEnv.ManagerRoleARN, o.targetEnv.Region)
+	if err != nil {
+		return fmt.Errorf("get env session: %w", err)
+	}
+	mft, err := workloadManifest(&workloadManifestInput{
+		name:         o.wkldName,
+		appName:      o.appName,
+		envName:      o.envName,
+		interpolator: o.newInterpolator(o.appName, o.envName),
+		ws:           o.ws,
+		unmarshal:    o.unmarshal,
+		sess:         envSess,
+	})
+	if err != nil {
+		return err
+	}
+	o.appliedDynamicMft = mft
+	o.repository, err = o.configureClients(o)
+	if err != nil {
+		return err
+	}
+
+	err = o.buildContainerImages(o)
+	if err != nil {
+		return err
+	}
+
+	containerNames := o.out.ContainerNames
+	imageNames := o.out.ImageNames
+	for i, image := range imageNames {
+		o.imageInfoList = append(o.imageInfoList, clideploy.ImagePerContainer{
+			ContainerName: containerNames[i],
+			ImageURI:      image,
+		})
+	}
+
+	var sidecarImageLocations []clideploy.ImagePerContainer //sidecarImageLocations has the image locations which are already built
+	manifestContent := o.appliedDynamicMft.Manifest()
+	switch t := manifestContent.(type) {
+	case *manifest.ScheduledJob:
+		sidecarImageLocations = getBuiltSidecarImageLocations(t.Sidecars)
+	case *manifest.LoadBalancedWebService:
+		sidecarImageLocations = getBuiltSidecarImageLocations(t.Sidecars)
+	case *manifest.WorkerService:
+		sidecarImageLocations = getBuiltSidecarImageLocations(t.Sidecars)
+	case *manifest.BackendService:
+		sidecarImageLocations = getBuiltSidecarImageLocations(t.Sidecars)
```

Comment:
i think this works?
```suggestion
	case *manifest.ScheduledJob, *manifest.LoadBalancedWebService, *manifest.WorkerService, *manifest.BackendService:
		sidecarImageLocations = getBuiltSidecarImageLocations(t.Sidecars)
```

Reply:
This think this will not work we get `t.Sidecars undefined (type any has no field or method Sidecars)`